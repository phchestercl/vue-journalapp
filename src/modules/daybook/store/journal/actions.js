/* export const myActions = async ({commit})=>{

} */
export const loadEntries = async (/* { commit } */)=>{
    

}

export const upDateEntry = async (/* { commit } */)=>{
    

}
export const createEntry = async (/* { commit } */)=>{
    

}