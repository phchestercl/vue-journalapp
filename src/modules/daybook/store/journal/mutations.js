/* export const myMutations =  (state)=>{
    

} */

export const setEntries =  (/* state */)=>{
    

}
export const upDateEntry =  (/* state */)=>{
    
}
export const addEntry =  (/* state */)=>{
    

}