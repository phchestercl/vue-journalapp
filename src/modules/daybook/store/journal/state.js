export default()=>({
    isLoading:true,
    entries:[
        {
            id:new Date().getTime(),
            date:new Date().toDateString(),
            text:'lElit non cupidatat id cupidatat ex cillum. Tempor Lorem nulla consectetur aliquip et Lorem sit. Non mollit aute reprehenderit Lorem adipisicing dolor fugiat sit nisi consequat proident voluptate non labore. Eiusmod occaecat adipisicing duis voluptate cupidatat in cupidatat voluptate. Consequat amet sunt deserunt anim amet nulla deserunt esse. Labore mollit enim mollit incididunt cillum ea ex ad.',
            picture:null
        },
        {
            id:new Date().getTime()+1000,
            date:new Date().toDateString(),
            text:'Incididunt aliqua est velit elit sunt minim et dolor aliqua amet ea. Laboris ea mollit adipisicing amet ad ad. Ea fugiat ad ipsum sunt esse. Adipisicing pariatur fugiat irure anim voluptate ipsum cillum dolor mollit.',
            picture:null
        },
        {
            id:new Date().getTime()+2000,
            date:new Date().toDateString(),
            text:'Irure do pariatur eu fugiat. Commodo exercitation do nulla mollit id aliqua et proident sit. Labore tempor cillum in velit quis. Cupidatat enim enim mollit dolore amet ad.',
            picture:null
        }
]
})