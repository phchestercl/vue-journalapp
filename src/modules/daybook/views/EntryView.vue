<template>
  <!-- <h1>Hola Mundo es una nueva entrada</h1> -->
  <div class="entry-title d-flex justify-content-between p-2">
      <div>
          <span class="text-success fs-3 fw-bold">15</span>
          <span class="mx-1 fs-3 ">Septiembre</span>
          <span class="mx-2 fs-4 fw-light">2021</span>

      </div>
      <div>
          <button class="btn btn-danger mx-2">
              Borrar
              <i class="fa fa-trash-alt"></i>
          </button>
           <button class="btn btn-primary ">
              Subir Foto
              <i class="fa fa-upload"></i>
          </button>
      </div>
  </div>
  <hr>
  <div class="d-flex flex-column px-3 h-75">
      <textarea placeholder="Que sucedio hoy"></textarea>

  </div>
  <Fab icon="fa-save"/>
  <img 
    src="https://www.autobild.es/sites/autobild.es/public/styles/855/public/dc/fotos/Volvo_S60_03.jpg?itok=1zT-Ffz-" 
    alt="Volvo"
    class="img-thumbnail">
</template>

<script>
import { defineAsyncComponent } from 'vue'
export default {
    components:{
        Fab:defineAsyncComponent(()=>import('../components/Fab.vue'))
    }

}
</script>

<style lang="scss" scoped>
    textarea{
        font-size: 20px;
        border:none;
        height: 100%;
        &:focus{
            outline: none;
        }
    }
    img{
        width: 200px;
        position: fixed;
        bottom: 150px;
        right: 20px;
        box-shadow: 0px 5px 10px rgba($color:#000000,$alpha:0.2);
    }
</style>