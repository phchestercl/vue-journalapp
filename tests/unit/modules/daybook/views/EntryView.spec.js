import { createStore } from 'vuex'
import { shallowMount } from '@vue/test-utils'
import EntryView from '../../../../../src/modules/daybook/views/EntryView.vue'
import journal from '@/modules/daybook/store/journal'
import { journalState } from '../../../mock-data/test-journal-state'
import Swal from "sweetalert2";
const createVuexStore = (initialSate) => createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialSate }
        }
    }
}
)
jest.mock('sweetalert2',()=>({
    fire:jest.fn() ,
    showLoading:jest.fn(),
    close:jest.fn()
}
))

describe('EntryList componnet', () => {
    //console.log(journal)
    let mockRouter
    let store
    store = createVuexStore(journalState)
    store.dispatch=jest.fn()   //==> con esto evitamos que el dispatch no se ejencute

    mockRouter = {
        push: jest.fn()
    }

    test('debe sacar al usuario por que al id no existe', () => {
        const wrapper = shallowMount(EntryView, {
            props:{id:'este-id-no-existe'},
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store]
            }
        })
        expect(mockRouter.push).toHaveBeenLastCalledWith({ name: "no-entry" })
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('Debe mostrar la entrada correctamente',()=>{
        //6137f23f7f283c228118d7b2
        const wrapper = shallowMount(EntryView, {
            props:{id:'6137f23f7f283c228118d7b2'},
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store]
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.vm.entry).toEqual(journalState.entries[0])
        
    })
    // test de borar la entrada y salir
    test('Borrar entrada y salir',(done)=>{
        // simular en click de borrado
        const wrapper = shallowMount(EntryView, {
            props:{id:'6137f23f7f283c228118d7b2'},
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store]
            }
        })
        // simula que Swal devuelve la respuesta con true
        Swal.fire.mockResolvedValueOnce( Promise.resolve({isConfirmed:true}))
        
        wrapper.find('.btn-danger').trigger('click')

        expect(Swal.fire).toHaveBeenCalledWith({
            title: "¿Está seguro de borrar la Orden de Trabajo?",
            text: "Una vez borrda, no se podrá recuperar.",
            showDenyButton: true,
            confirmButtonText: "Si, estoy seguro",
          })
          setTimeout(() => {
              expect(store.dispatch).toHaveBeenCalledWith("journal/deleteEntry", "6137f23f7f283c228118d7b2")
              expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" })
              done()
          }, 1);
          // simulacion y espia delete entry
    })



})
