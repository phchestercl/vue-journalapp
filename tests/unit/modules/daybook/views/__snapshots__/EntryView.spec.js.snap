// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryList componnet Debe mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">7</span><span class="mx-1 fs-3">Septiembre</span><span class="mx-2 fs-4 fw-light">2021, Martes</span></div>
  <div><input type="file" accept="image/*" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir Foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="Que sucedio hoy"></textarea></div>
<fab-stub icon="fa-save"></fab-stub>
<!-- <img
    src="https://www.autobild.es/sites/autobild.es/public/styles/855/public/dc/fotos/Volvo_S60_03.jpg?itok=1zT-Ffz-"
    alt="Volvo"
    class="img-thumbnail"
  /> -->
<!--v-if--><img src="https://res.cloudinary.com/philipcl/image/upload/v1631065477/cursovue/9-logo_wzsown.png" alt="entry-picture" class="img-thumbnail">
`;

exports[`EntryList componnet debe sacar al usuario por que al id no existe 1`] = `
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
<!-- <img
    src="https://www.autobild.es/sites/autobild.es/public/styles/855/public/dc/fotos/Volvo_S60_03.jpg?itok=1zT-Ffz-"
    alt="Volvo"
    class="img-thumbnail"
  /> -->
<!--v-if-->
<!--v-if-->
`;
